app/helpers/common_parameters_helper.rb

```diff
@@ -70,9 +70,24 @@ def lookup_key_field(id, value, options)
 
   def authorized_resource_parameters(resource, type)
     parameters_by_type = resource.send(type)
-    parameter_ids_to_view = parameters_by_type.authorized(:view_params).map(&:id)
-    resource_parameters = parameters_by_type.select { |p| parameter_ids_to_view.include?(p.id) }
+    user = User.current
+    @params_authorizer = Authorizer.new(user, :collection => parameters_by_type)
+    if user.nil? || user&.admin?
+      resource_parameters = apply_only_user_filter(user, parameters_by_type)
+    else
+      resource_parameters = @params_authorizer.find_collection(
+        parameters_by_type.klass, :permission => :view_params)
+    end
     resource_parameters += parameters_by_type.select(&:new_record?)
     resource_parameters
   end
+
+  def apply_only_user_filter(user, parameters_by_type)
+    return parameters_by_type.where('1=0') if user.nil?
```

Comment:
I would recommend to return `parameters_by_type.none` instead. It is an `ActiveRecord::Relation` acting as a Null Object that is still chainable, but does not generate a database query.